File: platform/fastify_server/config/lifecycle/530_github_sdk_api_component_usage_audit.lifecycle.mjs
/**
 * GitHub SDK API Component Usage Audit — Fastify Lifecycle Hook
 *
 * Registers a stub route for the component usage audit.
 * The actual audit logic runs through the Python/FastAPI server;
 * this hook is a placeholder for future Fastify-native implementation.
 *
 * Loading Order: 530 (after 500.github_sdk which provides the GitHub client)
 *
 * Registered endpoints:
 *   POST /~/api/rest/{api_release_date}/providers/github_sdk_api_component_usage_audit/v1/audit
 */

/**
 * Startup hook — Register component usage audit routes.
 *
 * @param {import('fastify').FastifyInstance} server
 * @param {object} config - Bootstrap config
 */
export async function onStartup(server, config) {
  server.log.info('[lifecycle:github_sdk_api_component_usage_audit] Initializing component usage audit routes...');

  try {
    // Verify that the GitHub SDK client was initialized
    if (!server.hasDecorator('github')) {
      server.log.warn(
        '[lifecycle:github_sdk_api_component_usage_audit] ' +
        'GitHub client not found — skipping route registration. ' +
        'Ensure 500.github_sdk lifecycle runs first.',
      );
      return;
    }

    server.log.info('[lifecycle:github_sdk_api_component_usage_audit] GitHub client found, proceeding with route registration');

    const apiReleaseDate = server.config?.getNested?.(
      ['api_release_date', 'contract_snapshot_date', 'provider_github'],
    ) ?? '02-01-2026';
    const PREFIX = `/~/api/rest/${apiReleaseDate}/providers/github_sdk_api_component_usage_audit/v1`;
    server.log.info({ PREFIX, apiReleaseDate }, '[lifecycle:github_sdk_api_component_usage_audit] Resolved route prefix');

    await server.register(
      async function componentUsageAuditRoutes(scope) {
        scope.post('/audit', async (request, reply) => {
          // Proxy to the FastAPI server for now
          const { componentName, minStars, maxPages, minFileSize } = request.body || {};

          if (!componentName) {
            return reply.code(400).send({ error: 'componentName is required' });
          }

          return reply.code(501).send({
            message: 'Component usage audit is served by the FastAPI server. ' +
              'POST to the FastAPI endpoint instead.',
            fastapi_endpoint: `${PREFIX}/audit`,
          });
        });

        return Promise.resolve();
      },
      { prefix: PREFIX },
    );

    server.log.info(
      { PREFIX },
      `[lifecycle:github_sdk_api_component_usage_audit] Routes registered at ${PREFIX}/*`,
    );
  } catch (err) {
    server.log.error({ err, hookName: '530_github_sdk_api_component_usage_audit' }, '[lifecycle:github_sdk_api_component_usage_audit] Component usage audit route registration failed');
    throw err;
  }
}

/**
 * Shutdown hook.
 *
 * @param {import('fastify').FastifyInstance} server
 */
export async function onShutdown(server) {
  server.log.info('[lifecycle:github_sdk_api_component_usage_audit] Shutdown complete');
}

File: platform/fastapi_server/config/lifecycle/530_github_sdk_api_component_usage_audit.lifecycle.py
"""
GitHub SDK API Component Usage Audit — FastAPI Lifecycle Hook

Registers the component usage audit route under the GitHub SDK API prefix.

Loading Order: 530 (after 500_github_sdk which provides the GitHub client)

Registered endpoints:
    POST /~/api/rest/{api_release_date}/providers/github_sdk_api_component_usage_audit/v1/audit
"""

import logging
import sys
from pathlib import Path
from typing import Any

from fastapi import APIRouter, FastAPI

# Add polyglot paths to sys.path
_root = Path(__file__).parent.parent.parent.parent
_module_path = _root / "polyglot" / "github_sdk_api_component_usage_audit" / "py"
_github_api_path = _root / "polyglot" / "github_api" / "py"
_packages_py_path = _root / "packages_py"

for _p in (_module_path, _github_api_path, _packages_py_path):
    if str(_p) not in sys.path:
        sys.path.insert(0, str(_p))

logger = logging.getLogger("lifecycle.github_sdk_api_component_usage_audit")


async def onStartup(app: FastAPI, config: dict[str, Any]) -> None:
    """Startup hook — Register component usage audit routes.

    Requires the GitHub SDK client to be initialized first (lifecycle 500).

    Args:
        app: FastAPI application instance.
        config: Bootstrap configuration dictionary.
    """
    logger.info("Starting github_sdk_api_component_usage_audit lifecycle hook...")
    try:
        # Check that github_client was initialized by 500_github_sdk
        client = getattr(app.state, "github_client", None)
        if client is None:
            logger.warning(
                "GitHub client not found on app.state — skipping route registration. "
                "Ensure 500_github_sdk lifecycle runs first."
            )
            return

        logger.debug("GitHub client found on app.state, registering audit routes")

        from github_sdk_api_component_usage_audit.routes.audit import create_router

        audit_router = create_router(client)

        # Build prefix
        api_release_date = app.state.config.get_nested(
            "api_release_date", "contract_snapshot_date", "provider_github"
        )
        PREFIX = (
            f"/~/api/rest/{api_release_date}"
            f"/providers/github_sdk_api_component_usage_audit/v1"
        )
        logger.debug("Component usage audit prefix: %s", PREFIX)

        parent_router = APIRouter(prefix=PREFIX)
        parent_router.include_router(audit_router)
        app.include_router(parent_router)

        logger.info("Routes registered at %s/*", PREFIX)
        logger.info("github_sdk_api_component_usage_audit lifecycle hook completed successfully")
    except Exception as exc:
        logger.error(
            "github_sdk_api_component_usage_audit lifecycle hook failed: %s", exc, exc_info=True
        )
        raise
